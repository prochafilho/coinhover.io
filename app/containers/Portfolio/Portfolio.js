import React, { Component } from 'react';
import { connect } from 'react-redux';

// Components
import { Header } from '../Common/Header';
import { CoinTable } from '../CoinTable/CoinTable';
import SocialMediaFooter from '../Common/SocialMediaFooter';

// Actions
import { getCoins } from '../../actions';

class Portfolio extends Component {
  componentDidMount() {
    this.props.getCoins();
  }

  render() {
    const { loading } = this.props;

    return (
      <div className="app-bg">
        <section className="portfolio">
          <Header />
          { loading ? (
            <div className="loading">
              <div className="loader" />
              <span>Loading coin data...</span>
            </div>
          ) : (
            <CoinTable />
          )}
          <SocialMediaFooter />
        </section>
      </div>
    );
  }
}

export const PortfolioJest = Portfolio;

const mapDispatchToProps = dispatch => ({
  getCoins: (...args) => dispatch(getCoins(...args))
});

const mapStateToProps = ({ coins, loading }) => ({
  coins,
  loading
});

export default connect(mapStateToProps, mapDispatchToProps)(Portfolio);
