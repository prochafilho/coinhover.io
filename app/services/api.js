import axios from 'axios';

const coinmarket = 'https://api.coinmarketcap.com/v1/ticker/';
const coinTicker = coinId => `https://api.coinmarketcap.com/v1/ticker/${coinId}/`;

const log = (method, err) => {
  /* eslint-disable no-console */
  console.warn(`%c${method}`, 'background: #393939; color: #F25A43', err);
  return err;
};

export const getAllCoins = () => axios.get(coinmarket)
  .catch(err => log('api.getAllCoins', err))
  .then(res => res);

export const getCoin = coinId => axios.get(coinTicker(coinId))
  .catch(err => log('api.getCoin', err))
  .then(res => res);
