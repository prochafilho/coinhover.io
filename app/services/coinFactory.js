/* eslint-disable no-param-reassign */
import * as R from 'ramda';

export const storage = {
  coins: [],
  portfolio: []
};

// const sortById = R.sortBy(R.compose(R.prop('id')));

const testMatch = (re, str) => str.search(re) !== -1;

export const findCoins = (text) => {
  const findMatches = coin => (testMatch(text, coin.name.toLowerCase()) ? coin : null);
  const matches = R.map(findMatches, storage.coins);
  return R.reject(R.isNil, matches);
};

const keysToClean = [
  'available_supply',
  'last_updated',
  'market_cap_usd',
  'max_supply',
  'price_btc',
  'total_supply'
];

// Clean Coins Function
export const cleanCoins = coins =>
  // Return our mapped coins array
  coins.map(coin =>
    // Iterate through each key in the object and create a new object (reduce)
    Object.keys(coin).reduce((newObj, key) => (
      // Check to see if this key is inside keysToClean
      keysToClean.indexOf(key) < 0
      // If not, add it to the new object
        ? ({ ...newObj, [key]: coin[key] })
        // Otherwise, ignore the key and move on
        // so that its not in our new object anymore
        : newObj
    ), {})
  );
