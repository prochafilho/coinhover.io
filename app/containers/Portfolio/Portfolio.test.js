import React from 'react';
import * as enzyme from 'enzyme';
import toJson from 'enzyme-to-json';
import Portfolio from './Portfolio';
import { updateLocalCoins } from '../../services/coinFactory';

const wrapper = enzyme.shallow(<Portfolio />);

describe('<Portfolio /> component', () => {
  it('should render', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();

    expect(wrapper).toHaveLength(1);
  });
});

describe('updateLocalCoins in coinFactory', () => {
  it('should be defined', () => {
    expect(updateLocalCoins).toBeDefined();
  });
});
