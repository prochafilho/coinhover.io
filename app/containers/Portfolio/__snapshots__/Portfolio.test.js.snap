// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Portfolio /> component should render 1`] = `
<div
  className="app-bg"
>
  <section
    className="portfolio"
  >
    <Header />
    <AssetsTable />
    <SocialMediaFooter />
  </section>
</div>
`;
